thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -1107,4 +1115,33 @@ public Response listYamls(
     LOG.info("Successfully returned " + yamls.size() + " detection configs.");
     return Response.ok(yamls).build();
   }
+
+  /**
+   * Api to trigger a notification alert. Mostly used for ad-hoc testing.
+   * Alert will be sent only if there are new anomalies since the last watermark.
+   * Watermarks will be updated after notifying anomalies if any.
+   */
+  @PUT
+  @Path("/notify/{id}")
+  @ApiOperation("Send notification email for detection alert config")
+  public void test(
+      @ApiParam("Subscription configuration id for the alert") @NotNull @PathParam("id") long detectionAlertConfigId) {
+    LOG.info("Triggering subscription task with id " + detectionAlertConfigId);
+
+    // Build the task context
+    ThirdEyeAnomalyConfiguration config = new ThirdEyeAnomalyConfiguration();
+    config.setAlerterConfiguration(alerterConfig);
+    TaskContext taskContext = new TaskContext();
+    taskContext.setThirdEyeAnomalyConfiguration(config);
+
+    // Run the notification task. This will update the subscription watermark as well.
+    DetectionAlertTaskInfo taskInfo = new DetectionAlertTaskInfo(detectionAlertConfigId);
+    TaskRunner taskRunner = new DetectionAlertTaskRunner();
+    try {
+      taskRunner.execute(taskInfo, taskContext);
+    } catch (Exception e) {
+      LOG.error("Exception while triggering the notification task with id " + detectionAlertConfigId, e);
```

Comment:
return the error message in the endpoint's response?

Reply:
Done